src/AnalysisEcce.cxx

```diff
@@ -177,15 +186,16 @@ void AnalysisEcce::Execute()
 
 	  // identify scattered electron by max momentum
 	  if(pid_ == 11) {
-	    if(p_ > maxP) {
-	      maxP = p_;
+	    if(pt_ > maxPt) {
+	      maxPt = pt_;
 	      kinTrue->vecElectron.SetPxPyPzE(px_, py_, pz_, e_);
 	      genEleID = part.mcID; //mcpart_ID[imc];
+	      genEleBCID = hepmcp_BCID[imc];
 	      //	      cout  << "\t\t\t found scattered electron  " << Form(" %6.2f %6.2f %6.2f %6.2f %6.2f  %5.3f %6.2f %6.2f id %3d\n",px_,py_,pz_, sqrt(p_*p_ + mass_*mass_),p_,mass_,hepmcp_E[imc],mcpart_E[imcpart],genEleID);
 	    }
```

Comment:
Should we switch to electron finding by max pT in the other `Analysis`-derived classes?